src/webui/components/Header/index.js

```diff
@@ -1,69 +1,179 @@
-                onClick={handleLogout}
-              >
-                Logout
-              </Button>
-            </div>
-          ) : (
-              <Button
-                className={`${classes.headerButton} header-button-login`}
-                color="primary"
-                onClick={toggleLoginModal}
-              >
-                Login
-            </Button>
-            )}
-        </div>
-      </div>
-    </header>
-  );
-};
-
-Header.propTypes = {
-  logo: PropTypes.string,
-  scope: PropTypes.string,
-  username: PropTypes.string,
-  handleLogout: PropTypes.func.isRequired,
-  toggleLoginModal: PropTypes.func.isRequired
-};
-
-export default Header;
+/**
+ * @prettier
+ */
+
+// @flow
+
+import React, {Component, MouseEvent} from 'react';
+import Button from '@material-ui/core/Button';
+import IconButton from '@material-ui/core/IconButton';
+import MenuItem from '@material-ui/core/MenuItem';
+import Menu from '@material-ui/core/Menu';
+import AccountCircle from '@material-ui/icons/AccountCircle';
+import Info from '@material-ui/icons/Info';
+import Help from '@material-ui/icons/Help';
+import FileCopy from '@material-ui/icons/FileCopy';
+import Tooltip from '@material-ui/core/Tooltip';
+
+import {getRegistryURL} from '../../utils/url';
+import {copyToClipBoard} from '../../utils/styles/mixins';
+import Link from '../Link';
+import Logo from '../Logo';
+
+import InfoDialog from './infoDialog';
+import {IProps, IState} from './interfaces';
+import {Wrapper, InnerWrapper, ClipBoardCopy, ClipBoardCopyText, CopyIcon} from './styles';
+
+class Header extends Component<IProps, IState> {
+  constructor() {
+    super();
+    this.handleMenu = this.handleMenu.bind(this);
+    this.handleClose = this.handleClose.bind(this);
+    this.handleOpenInfoDialog = this.handleOpenInfoDialog.bind(this);
+    this.handleCloseInfoDialog = this.handleCloseInfoDialog.bind(this);
+    this.handleToggleLogin = this.handleToggleLogin.bind(this);
+    this.renderInfoDialog = this.renderInfoDialog.bind(this);
+    this.state = {
+      anchorEl: null,
+      openInfoDialog: false,
+      registryUrl: '',
+    };
+  }
+
+  componentDidMount() {
+    const registryUrl = getRegistryURL();
+    this.setState({
+      registryUrl,
+    });
+  }
+
+  handleMenu(event: MouseEvent<HTMLElement>) {
+    this.setState({
+      anchorEl: event.currentTarget,
+    });
+  }
+
+  handleClose() {
+    this.setState({
+      anchorEl: null,
+    });
+  }
+
+  handleOpenInfoDialog() {
+    this.setState({
+      openInfoDialog: true,
+    });
+  }
+
+  handleCloseInfoDialog() {
+    this.setState({
+      openInfoDialog: false,
+    });
+  }
+
+  handleToggleLogin() {
+    const {toggleLoginModal} = this.props;
+    this.setState(
+      {
+        anchorEl: null,
+      },
+      () => toggleLoginModal()
+    );
+  }
+
+  renderLeftSide() {
+    return (
+      <Link to="/">
+        <Logo />
+      </Link>
+    );
+  }
+
+  renderRightSide() {

```

Comment:
Here we can do some separation. Keep login/logput flow separate. Icons can be grouped in other function.